tests/platform_tests/api/test_module.py

```diff
@@ -116,19 +127,31 @@ def test_get_presence(self, duthosts, enum_rand_one_per_hwsku_hostname, localhos
 
     def test_get_model(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, platform_api_conn):
 
+        duthost = duthosts[enum_rand_one_per_hwsku_hostname]
         for i in range(self.num_modules):
             if self.skip_absent_module(i,platform_api_conn):
                 continue
+
+            support_get_model = self.get_module_facts(duthost, True, "model")
```

Comment:
We did this as on a linecard/supervisor card in a T2 chassis, we have more than one module
For our linecards, chassis API object has 2 modules - SUPERVISOR0 and LINE-CARD.
```
admin@ixre-egl-board4:~$ show chassis modules status
       Name             Description    Physical-Slot    Oper-Status    Admin-Status
-----------  ----------------------  ---------------  -------------  --------------
 LINE-CARD3  Nokia-IXR7250E-36x400G                4         Online              up
SUPERVISOR0    Nokia-IXR7250-SUP-10                A         Online              up

```
For supervisor card, chassisAPI has modules for all the linecards, SFMs and supervisor card itself.

In Nokia's chassis, we don't support getting attributes of modules other than ours. Thus, the tests would fail. Since the same functionality regarding the attributes is achieved by the chassis API methods, we figured it would be easier to ignore getting attributes for any module.  For example, for get_serial, we have:

```
admin@ixre-egl-board4:~$ docker exec -it pmon bash
root@ixre-egl-board4:/# python3 
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sonic_platform
>>> p=sonic_platform.platform.Platform()
>>> c=p.get_chassis()
>>> c.get_num_modules()
2
>>> c.get_module(0).get_name()
'SUPERVISOR0'
>>> c.get_module(1).get_name()
'LINE-CARD3'
>>> c.get_module(1).get_serial()
'EAG2-02-028'
>>> c.get_serial()
'EAG2-02-028'
>>> 
```

Reply:
@harish-kalyanaraman How do we keep track of any module like line card or supervisor card after replacement if we don't have serial number/ part number /model number

Reply:
We can get the serial number / part number / model number for our own module - either by using 'c.get_module(<mod_num>.get_serial()' or 'c.get_serial' -  just not for other modules that are present in the chassis.

Reply:
I am not sure what you are asking. But, we can get the serial number / part number / model number for our own module - either by using 'c.get_module(<mod_num>.get_serial()' or 'c.get_serial' -where <mod_num> is the  just not for other modules that are present in the chassis.  Also, 'show platform syseeprom' on the module (linecard/supervisor card) will give the info as well.

Reply:
@harish-kalyanaraman I guess Prince's point is to not give a blind skip if the particular module don't have a model number. 
Can we do this 
   if the module name in this loop matches to the module under test ( like LC in case of linecard or SUP incase of supervisor ) make sure we have a model number -- else we can skip for other modules ( eg: skip  SUP in case of linecard as you mentioned above )

Reply:
@harish-kalyanaraman  can you take care of this ? thanks

Reply:
@judyjoseph - Do all other vendors support getting eeprom, base_mac, serial info for other modules. If so, we can look into enhancing Nokia device mgr implementation to support it.

Reply:
@sanmalho-git , I feel we are good as long as we know our module details, the testcase can validate that -- getting for all other modules in chassis, we can add as an enhancement. 

Include @Staphylo, @shyam77git  to comment for how it is implemented is in their platforms, we need to have a common approach.